pkg/provisioner/demo/demo.go

```diff
@@ -59,15 +76,12 @@ func (p *demoProvisioner) ValidateManagementAccess() (result provisioner.Result,
 
 	switch hostName {
 
-	case registrationErrorHost:
-		if p.host.SetErrorMessage("failed to register new host") {
-			p.log.Info("setting registration error")
-			p.publisher("RegistrationError", "Failed to register new host")
-			result.Dirty = true
-		}
+	case RegistrationErrorHost:
 		// We have set an error, so Reconcile() will stop
+		result.ErrorMessage = "failed to register new host"
+		p.log.Info("setting registration error")
```

Comment:
Was this change meant to be part of the next patch?
Does it even matter on GitHub?

Reply:
Ugh, maybe, but probably not. I should stop trying to decompose these things and just go with the PR flow, I guess.